src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -339,19 +217,13 @@ private void EventLoop()
                             context = null;
                             contextWrapper = default;
```

Comment:
Looking at comments; these are being nulled to Jit behaviour; `SocketAsyncContext? context` is defined outside of loop, so that should be moved to just above the loop. (not saying these statements should be removed, but the Jit might remove them anyway https://github.com/dotnet/runtime/issues/37064

However, this is an infinite loop that never leaves once its started, should it have `[MethodImpl(MethodImplOptions.AggressiveOptimization)]` or will it be Tier1-style complied on first Jit? /cc @kouvel

Reply:
@benaadams I suggest to deal with these concerns in #37138. Feedback is welcome there!

Reply:
> this is an infinite loop that never leaves once its started, should it have [MethodImpl(MethodImplOptions.AggressiveOptimization)] or will it be Tier1-style complied on first Jit?

Methods containing loops are currently not jitted at tier 0 by default. If there is prejitted code for the method then it would be used, otherwise it would be jitted with optimizations on first call. `AggressiveOptimization` also prevents the method from being prejitted when the module is R2R'ed, so for a long-running method like this one if it has beneficial cross-module inlining opportunities or if the jitted code would otherwise be better than prejitted code, the attribute may help. Since the changes in https://github.com/dotnet/runtime/pull/37138 introduce frequent calls to the hot parts of the method that are separated into a different method, adding the attribute to `EventLoop` may be less effective than before that change but depends.